### PR TITLE
Add benchmarks

### PR DESCRIPTION
**Pull request by cristaloleg**

### CODE REVIEW

File: builq_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+func BenchmarkBuildNaive(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		var bb builq.Builder
+		bb.Addf("SELECT %s FROM %s", "foo,bar", "table")
+		bb.Addf("WHERE id = %$", 123)
+		query, args, err := bb.Build()
+
+		switch {
```
Comment (junk1tm) [inline builq_bench_test.go:16]: Looks like a test to me, do we need it in benchmarks? If so, could be a testing helper due to all switches being exactly the same

**Review by junk1tm — APPROVED** [review]
LGTM 👍

File: builq_bench_test.go (hunk 1 above)
Comment (cristaloleg) [reply builq_bench_test.go:16]: Yes, it's needed, price is so low that it doesn't make sense to value it.

File: builq.go [hunk 2]
```diff
@@ -34,14 +34,23 @@ func (b *Builder) Addf(format string, args ...any) *Builder {
...
 
 	wargs := make([]any, len(args))
-	for i, arg := range args {
-		wargs[i] = &argument{value: arg, builder: b}
+	for i := range args {
+		wargs[i] = &argument{value: args[i], builder: b}
 	}
 
-	fmt.Fprintf(&b.query, format+"\n", wargs...)
+	_, err := fmt.Fprintf(&b.query, format, wargs...)
```
[the remark this answers is not in the document]
Comment (cristaloleg) [reply builq.go:48]: To no create new string for nothing.

File: GUIDE.md [hunk 3]
```diff
@@ -29,3 +29,20 @@ Both `%$` and `%?` formats can be extended with `+` or `#`:
 * `%#?` will expand slice of slices argument as `(?, ?), (?, ?), ... (?, ?)`
 
 Argument must be a slice (for `+`) or a slice of slices (for `#`), otherwise the `.Build()` method returns an error.
+
+## Speed
+
+Even with `fmt` packge speed is very good. If case you want zero-allocation query builder consider to cache query and just use it's value (works only for static queries like in `BenchmarkBuildCached`)
```
[the remark this answers is not in the document]
Comment (cristaloleg) [reply GUIDE.md]: Done, thx.

**Pull request merged**